omnigan/trainer.py

```diff
@@ -46,14 +45,40 @@ def __init__(self, opts, comet=False, verbose=0):
 
         self.device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
 
-        self.logger.exp = None
+        self.exp = None
         if comet:
-            self.logger.exp = Experiment(project_name="OmniGAN")
-            self.logger.exp.log_parameters(flatten_opts(opts))
+            self.exp = Experiment(project_name="omnigan", auto_metric_logging=False)
+            self.exp.log_parameters(flatten_opts(opts))
 
-    def log_losses(self):
-        # TODO
-        pass
+    def log_losses(self, model_to_update="G"):
+        """Logs metrics on comet.ml
+
+        Args:
+            model_to_update (str, optional): One of "G", "D" or "C". Defaults to "G".
+        """
+        if self.opts.train.log_level < 1:
+            return
+
+        if self.exp is None:
+            return
+
+        assert model_to_update in {
+            "G",
+            "D",
+            "C",
+        }, "unknown model to log losses {}".format(model_to_update)
+
+        losses = self.logger.losses.copy()
+        if self.opts.train.log_level == 1:
+            # Only log aggregated losses: delete other keys in losses
+            for k in self.logger.losses:
+                if k not in {"representation", "generator", "translation"}:
+                    del losses[k]
+        # convert losses into a single-level dictionnary
+        losses = flatten_opts(losses)
```

Comment:
@adrienju check this function's docstring to get what it does, it's quite simple